src/polyfem/solver/Optimizations.cpp

```diff
@@ -86,7 +88,21 @@ namespace polyfem::solver
 			for (const auto &arg : args)
 				forms.push_back(create_form(arg, var2sim, states));
 
+			// bool amips_form = false;
+			// for (auto f : forms)
+			// 	if (typeid(AMIPSForm) == typeid(*f.get()))
+			// 		amips_form = true;
+
+			// bool shape_parameter = false;
+			// for (auto v : var2sim)
+			// 	if (v->get_parameter_type() == ParameterType::Shape)
+			// 		shape_parameter = true;
+
+			// if (shape_parameter && !amips_form)
+			// 	log_and_throw_error("If optimizing a shape paramter, AMIPS form must be part of the objectives to prevent inverted elements!");
+
 			obj = std::make_shared<SumCompositeForm>(var2sim, forms);
+			obj->enable_energy_print("sum_of_objectives");
```

Comment:
why always enable?

Reply:
I think it's useful to have a print of the total sum in case you forget to add a print for an objective.

Reply:
okay but if you have two SumCompositeForm, they'll have the same name in the log

Reply:
This is true, but there is no other way (clean) to set the top level sum to have a print statement, especially since the sum objective is marked by a list and not an object. Open to suggestions though.